### PR TITLE
grip: 4.3.2 -> 4.4.0, patch to handle front matter

### PR DESCRIPTION
**Pull request by dtzWill**
Useful for local rendering of RFC's....

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -4895,18 +4895,27 @@ in {
   };
 
   grip = buildPythonPackage rec {
-    version = "4.3.2";
+    version = "4.4.0";
     name = "grip-${version}";
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: `pname`

**Review by dotlambda — COMMENTED** [review]
Would you mind moving the expression out of `python-packages.nix`?

**Comment by dtzWill** [comment]
Sure thing!

Or that's what I thought, but there's already a top-level attribute "grip" and I'm not sure what to call this.
(I was also changing it to use buildPythonApplication since it's an app..)

Thoughts?

**Comment by dotlambda** [comment]
Actually, I didn't mean moving it to all-packages.nix, just creating a default.nix in python-modules/grip/.
It is indeed an app and a library, see API section in their README.

**Comment by dtzWill** [comment]
Oh, well hooray then! Updated accordingly, thanks!

File: pkgs/development/python-modules/grip/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  patches = [
+    # Render "front matter", used in our RFC template and elsewhere
+    (fetchpatch {
+      url = https://github.com/joeyespo/grip/pull/249.patch;
+      sha256 = "07za5iymfv647dfrvi6hhj54a96hgjyarys51zbi08c51shqyzpg";
+    })
+  ];
+
+  buildInputs = [ pytest responses ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/grip/default.nix]: `checkInputs`

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.grip python3.pkgs.grip

**Comment by dtzWill** [comment]
D'oh, fixed. Feel free to squash (or I can do so, NP).  Thanks!

**Review by dotlambda — APPROVED** [review]
I can't even merge, feel free to do that :)

**Comment by dtzWill** [comment]
Since there's no rush on this and out of respect for code ownership, I'll defer to brave sir @FRidh.

Thanks for the review, one day I'll be good at python packaging PR's! :).

**Pull request merged**